src/GitHub.InlineReviews/Services/PullRequestSessionService.cs

```diff
@@ -88,11 +92,20 @@ public class PullRequestSessionService : IPullRequestSessionService
                 var chunks = DiffUtilities.ParseFragment(hunk);
                 var chunk = chunks.Last();
                 var diffLines = chunk.Lines.Reverse().Take(5).ToList();
+                var firstLine = diffLines.FirstOrDefault();
+                if (firstLine == null)
+                {
+                    log.Warning("Ignoring in-line comment in {RelativePath} with no diff line context", relativePath);
+                    continue;
+                }
+
+                var diffLineType = firstLine.Type;
                 var thread = new InlineCommentThreadModel(
                     relativePath,
                     comments.Key.Item1,
                     comments.Key.Item2,
                     diffLines,
+                    diffLineType,
```

Comment:
Do we need to pass `diffLineType` into the ctor separately, considering that at this point `firstLine != null`?

Reply:
I thought since we need to check for this situation anyway and have the data, we may as well pass it in. Keeps it a little simpler. What are the cons?

Reply:
Just duplicated information really - it makes the API less fault tolerant, because you can pass in a `diffLineType != diffLines[0]` and doing a `First()` on `diffLines` isn't going to be expensive. That's my feeling at least. Feel free to disagree though and I'll 👍.

Reply:
Makes sense. I didn't like `DiffLineType` failing when it came to be used, but there's no reason not to populate this in the constructor.